packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/totp_setup_state_machine.dart

```diff
@@ -59,29 +62,52 @@ final class TotpSetupStateMachine
         )
         .result;
     _session = response.session;
+    _details = TotpSetupDetails(
+      username: CognitoIdToken(tokens.idToken).username,
+      sharedSecret: response.secretCode!,
+    );
     emit(
       TotpSetupState.requiresVerification(
-        TotpSetupDetails(
-          username: CognitoIdToken(tokens.idToken).username,
-          sharedSecret: response.secretCode!,
-        ),
+        _details!,
       ),
     );
   }
 
   Future<void> _onVerify(TotpSetupVerify event) async {
     final tokens = await manager.getUserPoolTokens();
     final accessToken = tokens.accessToken.raw;
-    await _cognitoIdp
-        .verifySoftwareToken(
-          VerifySoftwareTokenRequest(
-            accessToken: accessToken,
-            session: _session,
-            userCode: event.code,
-            friendlyDeviceName: event.friendlyDeviceName,
+    try {
+      await _cognitoIdp
+          .verifySoftwareToken(
+            VerifySoftwareTokenRequest(
+              accessToken: accessToken,
+              session: _session,
+              userCode: event.code,
+              friendlyDeviceName: event.friendlyDeviceName,
+            ),
+          )
+          .result;
+    } on Exception catch (e, st) {
+      if (e is EnableSoftwareTokenMfaException && _details != null) {
+        logger.verbose(
+          'Failed to verify setup TOTP code. Retrying...',
+          e,
+        );
+        emit(
+          TotpSetupState.requiresVerification(
+            _details!,
           ),
-        )
-        .result;
+        );
+        return;
+      }
+      logger.error(
```

Comment:
We are catching and swallowing all exceptions other than `EnableSoftwareTokenMfaException`, correct? Is that what we want to do? I was expecting the exception to be rethrown.

Reply:
Good point. I made the exception set the state to failure which should bubble up the concern.